parts/22-query.sh

```diff
@@ -2406,93 +2410,307 @@ query_monthly-users-active(){ ## [year] [--by_group]: Number of active users per
-		     2018-06-01 | 230470
-		     2018-05-01 | 182331
-		     2018-04-01 | 109032
-		     2018-03-01 | 197125
-		     2018-02-01 | 260931
-		     2018-01-01 |  25378
+		YYYY-MM: Will return number of jobs for the given month
+
+		$ gxadmin query monthly-jobs 2024
+		  month  | count
+		---------+--------
+		 2024-02 |  71238
+		 2024-01 | 589359
 	EOF
 
+	state=""
+	group_by=""
+
 	if (( $# > 0 )); then
 		for args in "$@"; do
+			if [[ "$args" = "--by_state" ]]; then
+				state=", state"
+				group_by=", state"
+				continue
+			fi
 			if [[ "$args" = "--by_group" ]]; then
-				where_g="job.user_id = user_group_association.user_id and galaxy_group.id = user_group_association.group_id"
+				where_g="job.user_id = user_group_association.user_id AND galaxy_group.id = user_group_association.group_id"
 				select="galaxy_group.name,"
 				from="galaxy_group, user_group_association,"
 				group=", galaxy_group.name"
+			elif [[ "$args" =~ ^[0-9]{4}-[0-9]{2}$ ]]; then
+				where_m="date_trunc('month', job.create_time AT TIME ZONE 'UTC')::DATE = '$args-01'::date"
 			else
 				where_y="date_trunc('year', job.create_time AT TIME ZONE 'UTC') = '$args-01-01'::date"
 			fi
 		done
-		if (( $# > 1 )); then
-			where="WHERE $where_y and $where_g"
-		else
-			where="WHERE $where_y $where_g"
+		if [[ -n "$where_m" ]]; then
+			where="WHERE $where_m"
+			if [[ -n "$where_g" ]]; then
+			where="$where AND $where_g"
+		fi
+		elif [[ -n "$where_y" ]]; then
+			where="WHERE $where_y"
+			if [[ -n "$where_g" ]]; then
+				where="$where AND $where_g"
+			fi
+		elif [[ -n "$where_g" ]]; then
+			where="WHERE $where_g"
 		fi
 	fi
 
 	read -r -d '' QUERY <<-EOF
 		SELECT
-			date_trunc('month', job.create_time AT TIME ZONE 'UTC')::DATE AS month,
+			TO_CHAR(date_trunc('month', job.create_time AT TIME ZONE 'UTC')::DATE, 'YYYY-MM') AS month,
 			$select
 			count(*)
+			$state
 		FROM
 			$from
 			job
 		$where
 		GROUP BY
 			month
+			$state
 			$group
 		ORDER BY
 			month DESC
 	EOF
 }
 
-query_total-jobs(){ ## : Total number of jobs run by galaxy instance
+query_monthly-jobs-by-new-users() { ##? [month] [--no_state]: Number of jobs run by new users in the given month
 	meta <<-EOF
-		ADDED: 17
+		ADDED: 22
 	EOF
 	handle_help "$@" <<-EOF
-		Count total number of jobs
+		Count jobs run by users that registered in the given month
+		month: Month to count jobs for, provided as YYYY-MM. If month is not provided, defaults to current month.
+		--no_state: Do not break down jobs by state
+
+			$ gxadmin query jobs-by-new-users 2024-02
+			  state   | jobs_by_new_users
+			----------+-------------------
+			 deleted  |               340
+			 deleting |                 2
+			 error    |              1092
+			 new      |                41
+			 ok       |              4688
+			 paused   |                87
+			 stopped  |                 1
+	EOF
+
+	state="state,"
+	group_by_order_by="GROUP BY month, state ORDER BY month, state"
+
+	if [ $# -eq 0 ]; then
```

Comment:
```suggestion
	if [[ -z "$arg_month" ]]; then
```

there is always the chance I mess with which args are passed to functions, so checking the precise variable will be safer.